src/openms/source/APPLICATIONS/TOPPBase.cpp

```diff
@@ -1609,6 +1627,36 @@ namespace OpenMS
     }
   }
 
+  int TOPPBase::runExternalProcess_(const QString& executable, const QStringList& arguments) const
+  {
+    QProcess qp;
+    qp.start(executable, arguments); // does automatic escaping etc... start
+    std::stringstream ss;
+    ss << "COMMAND: " << executable.toStdString();
+    for (QStringList::const_iterator it = arguments.begin(); it != arguments.end(); ++it)
+    {
+        ss << " " << it->toStdString();
+    }
+    LOG_DEBUG << ss.str() << endl;
+    writeLog_("Executing: " + executable.toStdString());
+    const bool success = qp.waitForFinished(-1); // wait till job is finished
+    qp.close();
+
+    if (success == false || qp.exitStatus() != 0 || qp.exitCode() != 0)
+    {
+      writeLog_("FATAL: External invocation of " + executable.toStdString() + " failed. Standard output and error were:");
+      const QString stdout(qp.readAllStandardOutput());
+      const QString stderr(qp.readAllStandardError());
+      writeLog_(stdout);
+      writeLog_(stderr);
+      writeLog_(String(qp.exitCode()));
+
+      return EXTERNAL_PROGRAM_ERROR;
```

Comment:
do you catch this error?

Reply:
puh.. no.. I just want the program to exit. I see that it does not work anymore.

Reply:
yes, but now this returns from the function, not from `main` any more